.github/workflows/pull-request-workflow.yml

```diff
@@ -0,0 +1,37 @@
+name: Pull Request Workflow
+on: [pull_request]
+jobs:
+  run-tests:
+    runs-on: macos-11
+    timeout-minutes: 15
+    steps:
+    - name: Cancel previous jobs
+      uses: styfle/cancel-workflow-action@0.6.0
+      with:
+        access_token: ${{ github.token }}
+    - name: Git checkout
+      uses: actions/checkout@v2.3.4
+      with:
+        ref: ${{ github.ref }}
+    - name: Setup Xcode
+      uses: maxim-lobanov/setup-xcode@v1
+      with:
+        xcode-version: '12.4'
```

Comment:
Should we consider using 12.5.1 instead?

Reply:
Good question @antoniostrijdom
Sadly 12.5.1 is not yet available. See https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md#xcode